ega-data-api-htsget/src/main/java/eu/elixir/ega/ebi/htsget/service/internal/TicketServiceV2Impl.java

```diff
@@ -0,0 +1,203 @@
+        }
+
+        if (start.isPresent()) {
+            if (!referenceName.isPresent())
+                throw new InvalidInputException("Must specify reference name if using start parameter");
+
+            if (referenceName.get().equals("*"))
+                throw new InvalidInputException("Cannot use start with unplaced unmapped reads");
+        }
+
+        // Ascertain Access Permissions for specified File ID
+        File reqFile = fileInfoService.getFileInfo(id);
+
+        // Start Building the URL for the file on DataEdge
+        URI baseURI = UriComponentsBuilder
+                .fromUriString(externalConfig.getEgaExternalUrl() + id)
+                .queryParam("destinationFormat", "plain")
+                .build().toUri();
+
+        HtsgetResponseV2 result = new HtsgetResponseV2(format);
+
+        if (!onlyHeader && !referenceName.isPresent()) {
+            // user is requesting the entire file
+            result.addUrl(new HtsgetUrlV2(baseURI));
+            if (reqFile.getUnencryptedChecksumType() != null && reqFile.getUnencryptedChecksumType().equalsIgnoreCase("md5"))
+                result.setMd5(reqFile.getUnencryptedChecksum());
+
+        } else {
+
+            DataProvider reader = dataProviderFactory.getProviderForFormat(format);
+            if (!reader.supportsFileType(reqFile.getFileName()))
+                throw new UnsupportedFormatException("Conversion not supported");
+
+            try (SeekableStream dataStream = resClient.getStreamForFile(id)) {
+                reader.readHeader(dataStream);
+                result.addUrl(new HtsgetUrlV2(reader.getHeaderAsDataUri(), "header"));
+
+                if (!onlyHeader) {
+
+                    if (referenceName.get().equals("*")) {
+                        // TODO
+                        throw new NotImplementedException("Unplaced unmapped reads not implemented yet");
+                    }
+
+                    // Download the indexFile as we are going to look at it
+                    FileIndexFile fileIndexFile = fileInfoService.getFileIndexFile(reqFile.getFileId());
+                    if (fileIndexFile == null || StringUtils.isEmpty(fileIndexFile.getIndexFileId())) {
+                        throw new IndexNotFoundException("IndexFileId not found for file", id);
+                    }
+
+                    try (SeekableStream indexStream = resClient.getStreamForFile(fileIndexFile.getIndexFileId())) {
+                        reader.addContentUris(referenceName.get(), start.orElse(0L), end.orElse(Long.MAX_VALUE), baseURI, result, dataStream, indexStream);
+                    }
+
+                    result.addUrl(new HtsgetUrlV2(reader.getFooterAsDataUri(), "body"));
+                }
+                splitLargeDataBlocks(result);
+
+            }
+        }
+
+        return result;
+    }
+
+    private void splitLargeDataBlocks(HtsgetResponseV2 result) {
```

Comment:
Modify to accept the list or urls and return the new list of urls. This would additionally make some of the checkings not needed i.e. if all the urls are data urls.